larq_compute_engine/mlir/tf_tfl_passes.cc

```diff
@@ -92,6 +92,10 @@ void AddTFToLCETFLConversionPasses(
   pass_manager->addPass(
       mlir::tf_saved_model::CreateOptimizeGlobalTensorsPass());
 
+  // Set the batch size of the function input to 1 and let shape inference
+  // propagate this in the next pass.
+  pass_manager->addPass(mlir::CreateSetBatchSizePass());
```

Comment:
Originally I just applied it outside of the pass manager, directly on the `module` object that we get from `ImportSavedModel`. Then I decided a pass was cleaner because it also allows simple FileCheck tests.

First I placed the pass right before the first `tf-shape-inference` pass, but at that point there were several `@func`'s in the IR with these `StateFullPartitionedCall`s in them, so it was not clear whether this fix should be applied to all functions or not.
Then I placed it right before the second `tf-shape-inference` pass, at which point the `StateFullPartitionedCall`s had dissapeared and things were in the TF dialect.

So in conclusion: it can probably be placed in many locations, but the earlier the better since then more shapes will be known and more optimizations can be applied.

Reply:
Sounds good! Thanks for the explanation!